src/ddmd/cppmangle.d

```diff
@@ -34,225 +34,267 @@ import ddmd.visitor;
 /* Do mangling for C++ linkage.
  * Follows Itanium C++ ABI 1.86 section 5.1
  * http://refspecs.linux-foundation.org/cxxabi-1.86.html#mangling
+ * which is where the grammar comments come from.
  */
 
 extern (C++):
 
-extern (C++) const(char)* toCppMangleItanium(Dsymbol s)
+const(char)* toCppMangleItanium(Dsymbol s)
 {
     //printf("toCppMangleItanium(%s)\n", s.toChars());
-    scope CppMangleVisitor v = new CppMangleVisitor();
-    return v.mangleOf(s);
+    OutBuffer buf;
+    Target.prefixName(&buf, LINKcpp);
```

Comment:
Moving this call here.  Rather than being inside mangleOf.

Reply:
I moved it because it should be done first, not last, as doing it last requires rippling the mangled name up in the buffer. It's not an ABI change.

Reply:
Yep, I did make a mental note that it avoids a realloc/memmove.